### ISSUE TITLE
[Suggestion] Hovercraft changes.

### ISSUE THREAD

**Issue by DestinySilver** [tsunamayo/Starship-EVO#2216]
I was messing around with some hovercraft and I noticed a few things. First of all, narrow hovercraft crash a bit too easily, and they become excessively difficult to roll back over. There is a function to detect when the hovercraft is upside-down, so maybe add a feature that makes the hovercraft more likely to stop rolling while it is upright so it doesn't just roll all the way over and go right back to being upside down? It would certainly help some players' sanity.

Second, and far more important, 90 meters per second is not all that fast, in the grand scheme of things. Yeah, a collision at that speed will kill a person, but that speed should be nowhere near fast enough to hit a bump and send a hovercraft flying into space. I actually experienced this issue several times as I was testing things - running across a small hill which instead of acting as a ramp to send my hovercraft airborne for a few seconds, turned the craft's nose straight up and sent me sailing so far that I not only left the habitat atmosphere skybox effect, but actually sailed straight by the far side of the habitat and kept going - as far as 10 kilometers past this point, in fact. In one very special instance, I managed to get the angle of ascent right so that I actually hit the other side of the habitat and bounced up nearly out of the skybox again - showing just how ridiculous this ramp effect is. It seems to be a bug, but it could be just the way the game mechanics handle certain slopes.

Lastly, hovercraft seem to get stuck when they land underwater. They stall out, which seems to be an intentional game mechanic, but something that definitely isn't intentional is the way that the bikes start shaking violently - to such an extreme degree that they become impossible to push or move even with the gravity push tool, resulting in the craft being permanently stuck underwater. Worse yet, upon unrendering a hovercraft it will clip through the ground and appear inside of the intangible box on the outside of the habitat. Upon shoving it back upwards through the ground, the hovercraft in question will either resume it's shaking from before, if you pushed it into water, or clip through the map and wind up several thousand kilometers away. I don't have videos or pictures of any of this stuff, because sadly, I do not have a screen recorder, and a single frame wouldn't represent any of these problems very well.

**Comment by tsunamayo** [comment]
Seems there is two bugs for me!
For the first one I will implement something similar to the gravity gun in Garry's mod after suggestion from the community. 
Thanks

**Comment by Brushes2** [comment]
How do you feel about allowing directional thrusters on hovercraft? I've noticed that the tutorial examples tend to pitch up or down a lot and there just isnt a pitch control available for them, even if thrusters are present for those axis.

**Comment by DestinySilver** [comment]
I think that that would be really helpful. Obviously they are hovercraft and not full on spaceships, but there should definitely be some way to adjust your pitch and roll to some extent, for if you want to do aerial stunts, for example, or simply to correct trajectory.

**Comment by tsunamayo** [comment]
Yeah but that would mean completely remapping the control... Like you said they are not supposed to be aircraft, more a car like thing. That being said you can adjust roll axis.

**Comment by Brushes2** [comment]
Seems to me the only difference between the hover craft and the spaceships is that one stays close to the ground. They don't need vertical thrust (aside from the hover engines) but vertical thrusters could be used to provide some stability in pitch because, currently, they just end up crashing due to pitch up/down.

Maybe the hover computer could provide some magic pitch authority/stability instead?

**Comment by tsunamayo** [comment]
There is already some magic pitch stability, it is called anti-pitch in the Hovercraft computer settings.

**Comment by Brushes2** [comment]
It helps but isnt quite strong enough in my opinion.
